modules/partition-replicator/src/integrationTest/java/org/apache/ignite/internal/partition/replicator/ItReplicaLifecycleTest.java

```diff
@@ -582,6 +589,33 @@ void testAlterFilterTrigger(TestInfo testInfo) throws Exception {
         );
     }
 
+    @Test
+    void testTableReplicaListenersCreationAfterRebalance(TestInfo testInfo) throws Exception {
+        startNodes(testInfo, 3);
+
+        Assignment replicaAssignment = (Assignment) AffinityUtils.calculateAssignmentForPartition(
+                nodes.values().stream().map(n -> n.name).collect(Collectors.toList()), 0, 1).toArray()[0];
+
+        Node node = getNode(replicaAssignment.consistentId());
+
+        placementDriver.setPrimary(node.clusterService.topologyService().localMember());
+
+        DistributionZonesTestUtil.createZone(node.catalogManager, "test_zone", 1, 1);
+
+        createTable(node, "test_zone", "test_table");
```

Comment:
I would expect this check before table creation.
```
assertTrue(waitForCondition(() -> assertTableListenersCount(node, zoneId, 0), 10_000L));

```

Reply:
Added